### PR TITLE
ui: Test using Headless Chrome

### PR DESCRIPTION
**Pull request by mrtracy**
Begin running Karma tests using headless chrome instead of JSDOM. This
should ensure that our tests are running in the closest possible
environment to an actual browser.

Release note: None

### CODE REVIEW

**Comment by mrtracy** [comment]
Ran into yet another difference between jsdom and the browser (or rather, chrome and the browser) while working on some new networking tests.  Rather than try to debug that very obscure issue, we can simply make it a non-issue by moving our tests to use Headless Chrome instead of JSDom.

This seems like something we definitely want to do.

**Comment by vilterp** [comment]
Seems like a good idea 👍

**Comment by mrtracy** [comment]
Eh, this might not be a good idea because I don't think we can do this without making chrome a prequisite of our build.  I might try to find a way to fix the problem without doing this.

**Comment by mrtracy** [comment]
After talking with our build experts, we are fine with requiring the dependency on chrome, as it does not affect our non-ui build process.  Updated the README to reflect the new dependency.

This is ready for review.

File: pkg/ui/package.json [hunk 1]
```diff
@@ -84,8 +84,7 @@
     "jsdom": "^9.12.0",
```
Comment (couchand) [inline pkg/ui/package.json]: can we drop `jsdom`, too?

**Comment by benesch** [comment]
:lgtm:

I tested it too and it works magnificently.

---

Reviewed 4 of 6 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/25140#-:-LBX6beU7lmsbDxiA_Sp:bgkoaqg)*

**Comment by couchand** [comment]
I was expecting it to slow down `make test` by a lot, but it's only about 100ms, yay! 🎊

**Pull request merged**